targets/TARGET_STM/hal_tick_overrides.c

```diff
@@ -47,26 +46,22 @@ HAL_StatusTypeDef HAL_InitTick(uint32_t TickPriority)
 
 uint32_t HAL_GetTick()
 {
-#if TIM_MST_BIT_WIDTH == 16
-    uint32_t new_time;
-    if (mbed_sdk_inited) {
-        // Apply the latest time recorded just before the sdk is inited
-        new_time = ticker_read_us(get_us_ticker_data()) + prev_time;
-        prev_time = 0; // Use this time only once
-        return (new_time / 1000);
+    uint32_t new_time = us_ticker_read();
+    uint32_t elapsed_ticks = 0;
+    uint32_t elapsed_time = (((new_time - prev_time) & US_TICKER_MASK) + prev_tick_remainder);
+    prev_time = new_time;
+    if (TIM_MST_BIT_WIDTH <= 16 || elapsed_time < 65536) {
+        while (elapsed_time >= 1000) {
+            elapsed_ticks++;
+            elapsed_time -= 1000;
+        }
+        prev_tick_remainder = elapsed_time;
     } else {
-        new_time = us_ticker_read();
-        elapsed_time += (new_time - prev_time) & 0xFFFF; // Only use the lower 16 bits
-        prev_time = new_time;
-        return (elapsed_time / 1000);
+        elapsed_ticks = elapsed_ticks / 1000;
```

Comment:
That should be `elapsed_time / 1000`.

You would have got a warning about using an unset variable if you hadn't set `elapsed_ticks` to 0 at the start - good example of why bogus initialisation can cause errors. Personally I would do

    prev_time = new_time
    uint32_t elapsed_ticks;
    if (TIM_MST_BIT_WIDTH <= 16 || elapsed_time < 65536) {
        elapsed_ticks = 0;
        while (elapsed_time >= 1000) {

to maximally localise declaration and minimise unnecessary sets, but others' taste no doubt differs. The current form is fine once corrected, and probably compiles to the same code.